### PR TITLE
Wip adding localstorage

### PR DESCRIPTION
**Pull request by KrisGum**

### CODE REVIEW

File: index.html [hunk 1]
```diff
@@ -17,7 +19,10 @@ <h3 class="subtitle">Take a quiz to find your dream job!</h3>
             Do you want to be a programmer? Or maybe you have some experience and want to find a better job? Take a quiz to test your skills and find your dream job! At first, you have to choose your path. Then answer some questions to confirm your knowledge level and see the best job offers for you.
             </p>    
         </div> 
-        <button class="btn">Start</button>
-    </main>
+        <a href="./src/pages/path.html">
+            <button class="btn">Start</button>
+        </a>
```
Comment (jedrzejratajczak) [inline index.html]: If this is a link you should just use <a> tag and style it. <button> tags are for actions like validating form, showing modal. You don't need to use `onClick` so you should not use <button> tag.
https://medium.com/design-code-repository/a-vs-button-b859547cae4d

File: index.html [hunk 2]
```diff
@@ -5,9 +5,11 @@
     <meta http-equiv="X-UA-Compatible" content="IE=edge">
     <meta name="viewport" content="width=device-width, initial-scale=1.0">
     <title>Jobiz</title>
-
     <link rel="stylesheet" href="/src/assets/css/global.css">
+    <script async type="module" src="src/js/localstorage.js"></script> <!-- async so the localstorageTest will have priority-->
+    <!-- <script defer type="module" src="src/js/main.js"></script>        defer so the main.js will load symultanously -->
```
Comment (jedrzejratajczak) [inline index.html]: Why this code is commented? Do we need it? If we need it then why we commented this? File `main.js` do not exist. Nobody but you will know the answers to these questions and nobody will know if it is save to delete those comments.
In general do not leave any comments unless it is extreamly necessary and probably in all cases you won't need to use comments as I explained when you can use comments in previous comments.
Do we really need to tell what `defer` or `async` do? If anyone don't know it then he will check the documentation. Later this comment can even move to the wrong place and cause false information.
When developing a group project you need to think about other teammates. They will read it and waste time on those comments.

File: index.html [hunk 3]
```diff
@@ -5,9 +5,11 @@
     <meta http-equiv="X-UA-Compatible" content="IE=edge">
     <meta name="viewport" content="width=device-width, initial-scale=1.0">
     <title>Jobiz</title>
-
     <link rel="stylesheet" href="/src/assets/css/global.css">
+    <script async type="module" src="src/js/localstorage.js"></script> <!-- async so the localstorageTest will have priority-->
```
Comment (jedrzejratajczak) [inline index.html]: We don't need to import it here. We don't use any of those functions in this page. We will only need those functions in JS, not HTML files.

File: src/js/localstorage.js [hunk 4]
```diff
@@ -0,0 +1,187 @@
+//TESTING LOCALSTORAGE WHETHER IT WORKS OR EXIST
```
Comment (jedrzejratajczak) [inline src/js/localstorage.js]: As I said in previous comment, we do not need to use comments at all. Instead name functions and variables so they will tel us what they do. This is crucial in software development.

File: src/js/localstorage.js [hunk 5]
```diff
@@ -0,0 +1,187 @@
+//TESTING LOCALSTORAGE WHETHER IT WORKS OR EXIST 
+export function localStorageTest() {                 //to avoid name collisions
+    
+    try {
+        localStorage.setItem("test","test");
+        localStorage.removeItem("test");
+        console.log('Local Storage Fine')
```
Comment (jedrzejratajczak) [inline src/js/localstorage.js]: We don't need console to talk to us. Console logs are fine for debugging, but they should be removed then.

File: src/js/localstorage.js [hunk 6]
```diff
@@ -0,0 +1,187 @@
...
+      document.querySelector("main").insertAdjacentHTML("beforeend",'<button class="btn-fail">Start</button>');
+      document.querySelector(".btn-fail").addEventListener("mouseover", function () {
+      document.querySelector("main").insertAdjacentHTML("beforeend",'<p class="err-p">Fix your local storage</p>');});
+      document.querySelector(".btn-fail").addEventListener("mouseout", function () {document.querySelector(".err-p").remove();});
+    }
+  };
+
+//SAVING TO AND LOADING FROM LOCALSTORAGE - universal function
+
+export function saveToLocalStorage(resource, positionName) {
```
Comment (jedrzejratajczak) [inline src/js/localstorage.js]: Try to stick to one style of defining functions. Once you write `const lockStart = () => ...` then you use `function saveToLocalStorage(....) { ... }` and then `const nextQuestion = function (...) { ... }`. Those are 3 different styles in one file

File: src/js/localstorage.js [hunk 7]
```diff
@@ -0,0 +1,187 @@
...
+saveToLocalStorage(scoreboard, "scoreboard");
+}
+
+const getScoreboard = () => {
+    return loadFromLocalStorage("scoreboard");
+};
+
+//API PARTICULAR FUNCTIONS
+
+const setQuestions = function (...ques){ //ques == questions, could be any type
```
Comment (jedrzejratajczak) [inline src/js/localstorage.js]: Do not name variable so nobody will be able to tell what is it by her name and then explain in the comment what it is.
You can just name the variable `questions`.
We do not need to tell what type it is. We should be able to tell that it is a list by reading a variable name.

File: src/js/localstorage.js [hunk 8]
```diff
@@ -0,0 +1,187 @@
...
+};
+
+const removeLife = function () {
+    const player = getPlayer();
+    localStorage.removeItem("player");
+    player.lives--;
+    saveToLocalStorage(player, "player");
+};
+
+const addScore = function (n, s) { //n - string, player name, s - number, score
```
Comment (jedrzejratajczak) [inline src/js/localstorage.js]: The same problem as in previous comments. `n` and `s` are terrible names for variables. Why not just call it `playerName` and `score`? After like 2 months, when you'll forgot, you will not be able te tell what arguments you have to pass to the function and nobody else will be able either.

File: src/js/localstorage.js [hunk 9]
```diff
@@ -0,0 +1,187 @@
...
+    setQuestions: setQuestions,
+    nextQuestion: nextQuestion,
+    setPath: setPath,
+    setTier: setTier,
+    setPlayerName: setPlayerName,
+    useSkip: useSkip,
+    addPoint: addPoint,
+    removeLife: removeLife,
+    addScore: addScore
+};
```
Comment (jedrzejratajczak) [inline src/js/localstorage.js]: We can export variables and functions right before their definition.
`export const someVariable = ....`
`export function someFunction(...) { ... }`

File: src/js/localstorage.js [hunk 10]
```diff
@@ -0,0 +1,187 @@
...
+
+
+
+
+
+
+
```
Comment (jedrzejratajczak) [inline src/js/localstorage.js]: Do we need this empty space?

File: src/pages/nickname.html [hunk 11]
```diff
@@ -12,7 +12,15 @@
...
+        <a href="question.html" style="text-decoration: none;color:inherit">
+        <button class="btn">Enter!</button></a>
     </main>
+    <script defer type="module">
+        import {saveToLocalStorage, loadFromLocalStorage} from "../js/localstorage.js"
+
+        const button = document.querySelector("button");
+        button.addEventListener("click", function () {
+        saveToLocalStorage(document.querySelector('input').value, "enterName");});
+    </script>
```
Comment (jedrzejratajczak) [inline src/pages/nickname.html]: As I told before, HTML files are to hold HTML, not JS. JS files are to hold JS.
Do this like you do it in `questionPage.js`

File: src/pages/nickname.html [hunk 12]
```diff
@@ -12,7 +12,15 @@
...
-        <button class="btn">Enter!</button>
+        <a href="question.html" style="text-decoration: none;color:inherit">
+        <button class="btn">Enter!</button></a>
     </main>
+    <script defer type="module">
+        import {saveToLocalStorage, loadFromLocalStorage} from "../js/localstorage.js"
+
+        const button = document.querySelector("button");
+        button.addEventListener("click", function () {
+        saveToLocalStorage(document.querySelector('input').value, "enterName");});
```
Comment (jedrzejratajczak) [inline src/pages/nickname.html]: You've created dedicated functions to write objects to localStorage. Use them instead. We will have to change the key name in every place if we use it like there if we would like to change it.

File: index.html (hunk 2 above)
Comment (KrisGum) [reply index.html]: Comment deleted.

File: index.html (hunk 3 above)
Comment (KrisGum) [reply index.html]: Done.

File: src/js/localstorage.js (hunk 5 above)
Comment (KrisGum) [reply src/js/localstorage.js]: Done.

File: src/js/localstorage.js (hunk 7 above)
Comment (KrisGum) [reply src/js/localstorage.js]: Done.

File: src/js/localstorage.js (hunk 8 above)
Comment (KrisGum) [reply src/js/localstorage.js]: Done.

File: src/js/localstorage.js (hunk 10 above)
Comment (KrisGum) [reply src/js/localstorage.js]: Done.

File: src/pages/nickname.html (hunk 11 above)
Comment (KrisGum) [reply src/pages/nickname.html]: Done.

File: src/pages/nickname.html (hunk 12 above)
Comment (KrisGum) [reply src/pages/nickname.html]: Done.

File: index.html (hunk 1 above)
Comment (KrisGum) [reply index.html]: Done. Placed: 
"<a href="./src/pages/path.html" class="btn">Start</a>" instead the above.

File: src/js/localstorage.js (hunk 9 above)
Comment (KrisGum) [reply src/js/localstorage.js]: We decided to leave it like this but in the shorter version.

File: src/js/localstorage.js (hunk 6 above)
Comment (KrisGum) [reply src/js/localstorage.js]: Done.

**Review by jedrzejratajczak — APPROVED** [review]

**Pull request merged**